### PR TITLE
LRQA-66242 | Take a screenshot of all the rendered pages in each tab of clay sample portlet.

### PR DESCRIPTION
**Pull request by mateusralv**
Background
Create automation tests for take a screenshot of all the rendered pages in each tab of clay sample portlet.

Test Plan
Given: clay sample portlet deployed on page
When: navigate to each page
And when: viewing each screenshot
Then: each nav section of clay sample portlet looks normal

JIRA Ticket:
[LRQA-66242](https://issues.liferay.com/browse/LRQA-66242)

### CODE REVIEW

**Comment by mateusralv** [comment]
@john-co

**Comment by john-co** [comment]
ci:test:frontend:nocompile

**Comment by john-co** [comment]
Just started reviewing :)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClaySmoke.testcase [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+	@description = "Check if each nav section of clay sample portlet looks normal from screenshot"
+	@priority = "3"
+	test CanRenderUI {
+		task ("Change tabs and take a screenshot") {
+			for (var tabs : list "Alerts,Badges,Buttons,Cards,Dropdowns,Form Elements,Icons,Labels,Links,Management Toolbars,Navigation Bars,Pagination Bars,Progress Bars,Stickers") {
+				Navigator.gotoNavTab(navTab = "${tabs}");
+
+				takeScreenshot();
+			}
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/clay/ClaySmoke.testcase:53]: Unfortunately, it doesn't look like this for loop is going to translate well for our poshi report.

image

The poshi report seems to only save the last screenshot from the loop.

Maybe you should try making a new macro that navigates and takes screenshot, then have the for loop call that macro. Check that your local poshi report shows all screenshots in each tab.

**Pull request closed without merging**